simulation/src/transportation/bird_mob.rs

```diff
@@ -0,0 +1,137 @@
+use crate::map::Map;
+use crate::map_dynamic::Itinerary;
+use crate::physics::{Collider, CollisionWorld, PhysicsGroup, PhysicsObject, Speed};
+use crate::utils::rand_provider::RandProvider;
+use crate::utils::resources::Resources;
+use crate::utils::time::GameTime;
+use crate::World;
+use common::rand::rand3;
+use egui_inspect::Inspect;
+use geom::{angle_lerpxy, Transform, Vec3};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Inspect)]
+pub struct BirdMob {
+    pub walking_speed: f32,
+    pub fly_anim: f32,
+}
+
+const PED_SIZE: f32 = 0.5;
+
+pub fn put_bird_in_coworld(coworld: &mut CollisionWorld, pos: Vec3) -> Collider {
+    Collider(coworld.insert(
+        pos.xy(),
+        PhysicsObject {
+            radius: PED_SIZE * 0.6,
+            group: PhysicsGroup::Unknown,
+            ..Default::default()
+        },
+    ))
+}
+
+impl BirdMob {
+    pub(crate) fn new(r: &mut RandProvider) -> Self {
+        Self {
+            walking_speed: (10.0 + r.next_f32() * 0.4),
+            fly_anim: 0.0,
+        }
+    }
+}
+
+pub fn bird_decision_system(world: &mut World, resources: &mut Resources) {
+    profiling::scope!("transportation::animal_decision_system");
+    let ra = &*resources.read();
+    let map = &*resources.read::<Map>();
+    let next_dests: Vec<Vec3> = map
+        .buildings()
+        .values()
+        .map(|building| building.door_pos.up(-building.height))
+        // .chain(
+        //     map.terrain
+        //         .trees
+        //         .objects()
+        //         .map(|tree| vec3(tree.1.pos.x, tree.1.pos.y, tree.1.size)),
+        // )
+        .collect();
```

Comment:
this is far too expensive to do every frame when there will be a lot of buildings. You probably need something that queries the next destinations only when the birds need it. And ideally using something faster like using map.query (at a location) or fetching a few random buildings.

Reply:
Overall, I think it would be better for now if the bird just hanged around the map in a smooth way (not always going straight) and not get on the buildings. That would be simpler overall.

Bonus points if they travel in groups :)